swift-sdk/AuthManager.swift

```diff
@@ -51,16 +56,43 @@ class AuthManager: IterableInternalAuthManagerProtocol {
     
     func retrieveAuthToken() {
         authToken = localStorage.authToken
+        
+        queueAuthTokenExpirationRefresh(authToken)
     }
     
     // MARK: - Private/Internal
     
+    private static let refreshWindow = 60
+    
+    private var expirationRefreshTimer: Timer?
+    
     private var authToken: String?
     
     private var localStorage: LocalStorageProtocol
     
     private let onAuthTokenRequestedCallback: (() -> String?)?
     
+    private func queueAuthTokenExpirationRefresh(_ authToken: String?) {
+        guard let authToken = authToken, let expirationDate = AuthManager.decodeExpirationDateFromAuthToken(authToken) else {
+            return
+        }
+        
+        let refreshDate = TimeInterval(expirationDate - AuthManager.refreshWindow)
+        
+        if #available(iOS 10.0, *) {
+            expirationRefreshTimer = Timer.scheduledTimer(withTimeInterval: refreshDate, repeats: false) { timer in
```

Comment:
`scheduledTimer` method expects time interval and not a date.

Reply:
I'm going to rename it, `refreshDate` is actually a `TimeInterval`.